int16 speed1
int16 speed2
